src/app/shared/provider.service.ts

```diff
@@ -24,25 +35,46 @@ export class ProviderService {
 
     tool.provider = this.getProvider(gitUrl);
     tool.providerUrl = this.getProviderUrl(gitUrl, tool.provider);
+    tool.providerIcon = this.getProviderIcon(tool.gitUrl);
     return tool;
   }
 
+  private getProviderIcon(providerUrl: string | null): IconDefinition | null {
```

Comment:
You already have the provider from line 25/36. Instead of calling `getProvider` again on line 43, just pass in the provider as a parameter here.

Reply:
Done.